keps/sig-testing/2420-reducing-kubernetes-build-maintenance/README.md

```diff
@@ -0,0 +1,671 @@
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. The KEP
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this enhancement:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha -> Beta Graduation
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Tests are in Testgrid and linked in KEP
+
+#### Beta -> GA Graduation
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+#### Removing a Deprecated Flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include 
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+-->
+
+This will be declared stable/GA when:
+- All kubernetes/kubernetes `master` branch CI jobs use the preexisting make build system
+<!-- TODO: as old jobs are rotated out each release anyhow, it may be acceptable to reduce the stable target to just the current development branch -->
+- Bazel-related source files and related tooling are removed from the kubernetes/kubernetes repository on currently supported release branches and the current development branch
+  - This will only happen for release branches as we phase out support for older releases, rotating in new supported releases that never contained the Bazel build
+- Bazel-related configuration/presets are removed from kubernetes/kubernetes CI jobs in kubernetes/test-infra
+
+
+As bazel-built artifacts are not built or distributed as part of a kubernetes/kubernetes release, there is no deprecation window required.
```

Comment:
I wrote the "no deprecation window" part, but in retrospect I want to be clear that this merits a visible announcement.  It's not clear to me that "deprecation" or "action required" are a fit here.
```suggestion
Removal of bazel-related source files from kubernetes/kubernetes will be announced visibly (e.g. in the CHANGELOG, notification to kubernetes-dev@). However, as bazel-built artifacts are not built or distributed as part of a kubernetes/kubernetes release, there is no deprecation window required.
```

Reply:
I'm OK addressing this in followup, sounds like @justaugustus has thoughts on improving how we would announce something like this (ref: https://github.com/kubernetes/community/issues/5344)